session.go

```diff
@@ -777,10 +806,19 @@ func (s *Session) closeStream(id uint32) {
 	if s.client == (id%2 == 0) {
 		s.numIncomingStreams--
 	}
-	delete(s.streams, id)
+	s.deleteStream(id)
 	s.streamLock.Unlock()
 }
 
+func (s *Session) deleteStream(id uint32) {
+	str, ok := s.streams[id]
+	if !ok {
+		return
+	}
+	s.memoryManager.ReleaseMemory(int(str.recvWindow))
```

Comment:
We could use a transactional scope here, but it's easy enough to just release the memory manually.